services/api/src/api/authentication.py

```diff
@@ -42,12 +45,20 @@ def auth_check(
           which will be replaced with the dataset name, for example: https://huggingface.co/api/datasets/%s/auth-check
           The authentication service must return 200, 401, 403 or 404.
           If None, the dataset is always authorized.
-        request (Request | None): the request which optionally bears authentication headers: "cookie" or
-          "authorization"
+        request (Request | None): the request which optionally bears authentication headers: "cookie",
+          "authorization" or "X-Api-Key"
+        external_auth_bypass_key (str|None): if the request bears this secret key in the "X-Api-Key" header, the
+          external authentication service is bypassed.
 
     Returns:
         None: the dataset is authorized for the request
     """
+    if (
+        external_auth_bypass_key is not None
```

Comment:
In the README it says this: 

> Defaults to `""`.

But here it checks against `None`

Reply:
```suggestion
        external_auth_bypass_key
```
```suggestion
        not not external_auth_bypass_key
```
I feel like either would be slightly better, to guard against the env being set to an empty string as well

Reply:
You're right, thanks!